gradle/testUtils.gradle

```diff
@@ -0,0 +1,122 @@
+
+    // transform file name to java class name
+    def tests = testFiles.collect { fileName ->
+        fileName.replaceAll("${projectDir}/src/test/java/", '').replaceAll('\\.java$', '').replaceAll('/', '.')
+    }
+    tests = tests.findAll { testName ->
+        testName != "water.AAA_PreCloudLock" &&
+                testName != "hex.AAA_PreCloudLock" &&
+                testName != "water.parser.ParseProgressTest" &&
+                testName != "water.fvec.WordCountBigTest"
+    }
+
+    def ignoreList = System.getenv('IGNORE')
+    if (ignoreList) {
+        // Remove ignored tests
+        tests = tests.findAll { testName ->
+            !ignoreList.contains(testName)
+        }
+    }
+
+    def doOnly = System.getenv('DOONLY')
+    if (doOnly) {
+        // Remove all except DOONLY
+        tests = tests.findAll { testName ->
+            doOnly.contains(testName)
+        }
+    }
+
+    tests.sort()
+    return tests
+}
+
+enum ExecutionModes {
+    SINGLE_NODE("single_node_${System.currentTimeMillis()}"),
+    MULTI_NODE("multi_node_${System.currentTimeMillis()}")
+
+    private final String clusterName
+
+    ExecutionModes(final clusterName) {
+        this.clusterName = clusterName
+    }
+
+    String getClusterName() {
+        return clusterName
+    }
+}
+
+def executeTests(final ExecutionModes mode) {
+    def tests = getTests()
+    println "Running ${project.name} multinode tests (${tests.size()} test classes)"
+    new File("${buildDir}/tests.txt").text = tests.join(System.lineSeparator())
+
+    // assemble command for command line
+    def cmd = ['java', '-Xmx3g', '-Xms3g', '-ea', '-cp', sourceSets.test.runtimeClasspath.asPath]
+    cmd += ["-Dbuild.id=${System.getenv('BUILD_ID') ?: ''}", "-Djob.name=${System.getenv('JOB_NAME') ?: ''}", "-Dgit.commit=${System.getenv('GIT_COMMIT') ?: ''}", "-Dgit.branch=${System.getenv('GIT_BRANCH') ?: ''}", "-Dai.h2o.name=junit_cluster_${mode.clusterName}", "-Dai.h2o.baseport=43000", "-Dai.h2o.ga_opt_out=yes", "water.junit.H2OTestRunner", "water.AAA_PreCloudLock"]
+    cmd += tests
+    // print command to file, for debugging
+    new File("${buildDir}/jvm_cmd_${mode}.txt").text = cmd.join(' ')
+    // execute tests
+    exec {
+        commandLine cmd
+        // redirect stdout to file
+        def fileOutputStream = new FileOutputStream("${buildDir}/h2o-tests-${mode}-log.out")
+        standardOutput = fileOutputStream
+        errorOutput = fileOutputStream
+    }
+}
+
+task testMultiNodeNew(dependsOn: ['prepareDaemonTasks', "startDaemon_${multiNodeClusterSize - 1}"]) {
+    doLast {
+        executeTests ExecutionModes.MULTI_NODE
+    }
+}
+
+task testSingleNodeNew(dependsOn: ['prepareDaemonTasks']) {
+    doLast {
+        executeTests ExecutionModes.SINGLE_NODE
+    }
+}
+
+task prepareDaemonTasks() {
+    delete "${projectDir}/sandbox"
+    delete fileTree(buildDir) { include '**/h2o_*.log', '**/h2o_*.err' }
+    file("${projectDir}/sandbox").mkdirs()
```

Comment:
can we instead create the sandbox under `build` directory, to reduce the source tree pollution?